ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/Witnesses.scala

```diff
@@ -10,86 +10,94 @@ import scalaz.Tag
 
 final class Witnesses(session: LedgerSession) extends LedgerTestSuite(session) {
 
-  val respectDisclosureRules =
+  private[this] val respectDisclosureRules =
     LedgerTest("RespectDisclosureRules", "The ledger should respect disclosure rules") {
       implicit context =>
         for {
           Vector(alice, bob, charlie) <- allocateParties(3)
 
-          // Create Witnesses contract
-          (witnessTxId, witnesses) <- create(WitnessesTemplate(alice, bob, charlie))(alice)
-          witnessTx <- transactionTreeById(witnessTxId, alice, bob, charlie)
+          // Create the Witnesses contract as Alice and get the resulting transaction as seen by all parties
+          (witnessesTransactionId, witnesses) <- createAndGetTransactionId(
+            WitnessesTemplate(alice, bob, charlie))(alice)
+          witnessesTransaction <- transactionTreeById(witnessesTransactionId, alice, bob, charlie)
 
-          // Divulge Witnesses contract to charlie, who's just an actor and thus cannot
-          // see it by default.
-          (_, divulgeWitness) <- create(DivulgeWitnesses(alice, charlie))(charlie)
-          _ <- exercise(divulgeWitness.contractId.exerciseDivulge(alice, witnesses.contractId))(
-            alice)
+          // Charlie is not a stakeholder of Witnesses and thus cannot see any such contract unless divulged.
+          // Such contract is divulged by creating a DivulgeWitness with Charlie as a stakeholder and exercising
```

Comment:
More specifically signatory:
```suggestion
          // Such contract is divulged by creating a DivulgeWitness with Charlie as a signatory and exercising
```